### ISSUE TITLE
Autority -subject

### ISSUE THREAD

**Issue by SykoraLukas** [proarc/proarc#1334]
Když se stahují autority do elementu mods:subject, je nějaké místo, kam zapsat identifikátor této autority? U jmenných autorit se stáhne autor a jeho identifikátor se zapisuje do nameIdentifier. 

Je nějaké pole kam zapisovat identifikátor u autorit v subjektu? Když už ho dostáváme, tak mi přijde, že je škoda aby se zahazoval, ale pokud je to tak správně (že se zahodí) tak to tak nechám.

Např. identifikátor ph1046613 se nikam nestáhne, stáhnou se jen pole Dějiny Francie a Monaka a pole Vojenství, Obrana zeme, Ozbrojene sily.
![image](https://user-images.githubusercontent.com/27210823/119636032-8aaaba00-be14-11eb-9d8e-9c8129a9de3f.png)

**Comment by SykoraLukas** [comment]
Mohl bych poprosit @nezbedova @luckajirku @tumovam, zda by zkusily zjistit, jestli můžeme někam zapsat identifikátor těchto autorit?

**Comment by luckajirku** [comment]
no v loc jsem našla nějaké příklady (s lcsh, ale přijde mi to takhle od pohledu použitelné i s jinými slovníky):
<subject authority="lcsh" valueURI= "http://id.loc.gov/authorities/subjects/**sh85069617**">
<topic authority="lcsh" valueURI="http://id.loc.gov/authorities/subjects/**sh85069615**">Japanese drama</topic>
<temporal authority="lcsh" valueURI="http://id.loc.gov/authorities/subjects/**sh2002012465**">Edo period, 1600-868</temporal>
</subject>

**Comment by luckajirku** [comment]
na loc jsem něco našla, co by myslím mělo jít použít i s jinými slovníky než jen lcsh. jen se mi to sem nedaří viditelně zkopírovat, jak tam jsou ty závorky:-(. dole v examples:
https://www.loc.gov/standards/mods/userguide/subject.html#examples

**Comment by SykoraLukas** [comment]
myslíš to, jak se na to odkazuje ve valueURI=...   to bohužel použít nejde, protože Aleph NKP nepodává URL adresu. A přes tu,  která bz se dala dostat, tak v ní zase není žádné číslo autority.

https://aleph.nkp.cz/F/S27XMC2CK6GLKJ3MFF2QL5UVEUD7MTHILTL9N7PCUPIYGI3M1K-30421?func=full-set-set&set_number=224091&set_entry=000001&format=001

**Comment by SykoraLukas** [comment]
Teoreticky to muzeme trochu ohnout. Do pole authority napsat "AUT NK ČR" a do pole authorityURI napsat číslo dané autority. Ale bylo by to bez odkazu na danou autoritu. NKP totiz odkazuje pomocí systemoveho cisla a ne pomoci cisla autority.

**Comment by luckajirku** [comment]
no ono když se podíváš na ta sysna a na čísla autorit, tak ten konec je stejný, liší se jen ten začátek (nuly vs ph)

**Comment by SykoraLukas** [comment]
jak kde, když vzhledám autoritu Astronomický ústav, tak jich to najde několik. Každý má jiný prefix a pak tu je dokonce jedno, jehož číslo autority nijak nesouvisí se systémovým číslem.

https://aleph.nkp.cz/F/HE251JX1QCD9S5318K8HIMTES7P84ENM7BBHSCKAKXX3UVQSTI-55881?func=find-b&find_code=WRD&x=0&y=0&request=Astronomick%C3%BD+%C3%BAstav&adjacent=N

**Comment by luckajirku** [comment]
aha, já myslela, že se teď bavíme o předmětových heslech. takže myslíš obecně u autorit, i jmenných?

**Comment by SykoraLukas** [comment]
Ne promin, to jsem se do toho zamotal já. Jmenné na to mají určený nameIdentifier.

**Comment by SykoraLukas** [comment]
Když na to koukám, da se říci, že geografický kod bude vždy s prefixem geo a věcná autorita s prefixem ph?

**Comment by luckajirku** [comment]
Jak jsem to prošla, tak mi nepřijde, že by v aktuálních šablonách atd. v LOC to někde bylo. Přitom mi to není jasné, proč mají vyřešený odkaz, ale to číslo samotné nikam zapsat nejde... Přišlo by mi logické mít to stejně jako je nameIdentifier - tak subjectIdentifier...

**Comment by SykoraLukas** [comment]
Jo přesně tak bych si to nějak představoval. Dá se tohle nějak dostat do standardu?

**Comment by luckajirku** [comment]
Tak jednak buď můžeme vzít přes NK, aby se poptali v LOC, jestli by to šlo přidat, nebo to můžeme vzít přes LOC sami a pokud to tam přidají, tak řešit standard až pak.

**Comment by SykoraLukas** [comment]
A mohla by tohle řešit nějaká velká knihovna? Hradec/KNAV/MZK? Když napíše Inovatika, tak se s ní nejspíš nikdo nebude bavit.

**Comment by ZdenkaSera** [comment]
Měla by řešit KNAV

**Comment by ZdenkaSera** [comment]
KNAV řeší s NK zde: https://github.com/NLCR/Standard_NDK/issues/190

**Comment by kerschfilip** [comment]
Zapisování identifikátoru je na úrovni standardu NDK dořešené takto:

> Použije se atribut valueURI pro možnost zápisu odkazu do databáze autorit ve formátu „https://aleph.nkp.cz/dai/číslo_autority“ s využitím identifikátoru autority.
> 
> Atribut authority nebude nutné pro subelementy vytvářet, autoritní báze bude stanovena v atributu authority přímo pod topelementem <subject>, jako doposud.

Například: 
``` xml
<subject authority="czenas">
    <topic valueURI="https://aleph.nkp.cz/dai/ph121664">knihovnictví</topic>
</subject>
```
Prosím tedy o implementaci do PA. Děkuju!

**Comment by SykoraLukas** [comment]
Nasazeno k testování i s novým klientem.

**Comment by tumovam** [comment]
Zapisování identifikátoru je v pořádku, zapisuje se jako valueURI="https://aleph.nkp.cz/dai/... 

Ale všimla jsem si, že se nám nedoplňuje <mods:subject authority="czenas"> příp. AJ <mods:subject authority="eczenas"> (v DMF je poznámka u "authority", že _při použití volných klíčových slov atribut authority nepoužívat_ , ale když stahujeme z katalogu Autority NKP, tak přece nejsou volně tvořená?), příp. prostě jazyk, ve kterém je dané klíčové slovo. 

Stejně tak mi nejde stáhnout subject v angličtině. Zkoušela jsem na masteru - Katalog Autority NKP - Any - Dotaz "ultraviolet radiation" -> výsledek "ultrafialové záření", příp. třeba "photosynthesis" --> najde 3 výsledky, nic v AJ, jedno je "fotosyntéza".

**Comment by SykoraLukas** [comment]
To s tou autoritou bude problem. Autorita se přebírá z pole 072 podpole 2. 

Termíny, které stahujeme jsou v poli 150 pro topic. Tam není žádná vazba na slovník. Má se tedy přebírat authorita do slovníku z pole 072 podpole 2?

https://aleph.nkp.cz/F/?func=direct&doc_number=000120339&local_base=AUT

**Comment by SykoraLukas** [comment]
Řekl bych ale, že to motáme dohromady. V poli 150 není žádná informace o tom, že by to byl slovníkový termín. Tedy mi přijde správné, že to je bez autority.

Slovníkový termín je v poli 072 a proto má u sebe zapsanou autority.

**Comment by tumovam** [comment]
Jo, asi už to motám dohromady. Já jsem to brala i s ohledem na Filipův příklad posvěcený NKP :) https://github.com/proarc/proarc/issues/1334#issuecomment-2480765984

**Comment by SykoraLukas** [comment]
OK.

K tomu anglickému hledání:

![image](https://github.com/user-attachments/assets/e7af1d8e-dffd-4be8-b7d5-4df90d4e9481)

Problém je, že anglický výraz je zapsaný jen jako ekvivalent, ale záznam je v jazyce čeština. Proto i záznamy, které se vrátí jsou v češtině, i když hledáš v angličtině.

![image](https://github.com/user-attachments/assets/f290eed8-cb09-4639-b1a9-718b6843b417)

**Comment by SykoraLukas** [comment]
hodnoty v poli 072 jsou slovníkové (hodnota 072/x a slovník 072/2). V anglick0m ekvivalentu je ale v poli 750 uvedeno podpole 2 s hodnotou eczenas. 

Jestli ale máme všechny autority v češtině a anglicky jsou jen ekvivalenty, tak můžeme nastavit, že se bude automaticky přidávat hodnota czenas do autority. 

Bylo by to tak v pořádku?

**Comment by SykoraLukas** [comment]
Doplněno podle příkladu od NK

**Comment by tumovam** [comment]
Zkusme to prosím tak, že ke staženému subjectu v češtině přidej "czenas". To vyhledávání anglických ekvivalentů musíme probrat s jinými kolegy.

**Comment by tumovam** [comment]
Ve v4.2.4 se po stažení české autority nepropisuje "czenas". Jinak valueURI="https://aleph.nkp.cz/dai/... v pořádku.

Plus mám dotaz k řazení, lze nějak ovlivnit, nebo nám jen rexlektuješ to, co dostáváš k NKP a nelze to nijak seřadit (teba abecedně, nebo podle relevance)? 
![subject](https://github.com/user-attachments/assets/bafe20bb-3dc2-43ed-bb08-2619f5182744)
Díky

**Comment by SykoraLukas** [comment]
Podívám se.

Řazení je dle NK. U dalších přidaných záznamů autorit se přidává vždy na poslední místo.

**Comment by tumovam** [comment]
Na schůzce domluveno, že u českých klíčových slov, které se budou stahovat, se bude přidávat authority="czenas".

Anglické, ani jiné cizojazyčné ekvivalenty nebudeme řešit. Do PA se cizojazyčné ekvivalenty nebudou stahovat, ale pouze volně tvořit. Takže prosím neřešit komentáře výše o ""eczenas". Díky moc

**Comment by albertoh** [comment]
Nasazeno. @SykoraLukas muzes to otestovat?

**Comment by SykoraLukas** [comment]
Funguje, díky.

Jde o funkci, která se spouští na API, proto pokud je cokoli editované, je políčko pro přidání autorit (name/subject) zašedlé. Po uložení/vrácení změn se funkce zaktivní. Po přidání autority se refreshne formulář do kterého se přidá nová autorita.

**Comment by SykoraLukas** [comment]
K testování.

**Comment by tumovam** [comment]
Přidat s jazykem v pořádku. Stažené autority mají valueURI i "authority="czenas" 👍 , ale narazili jsme na zašedlý/neaktivní symbol při načítání metadat z katalogu.

Zpracovatel většinou kombinuje stahování a volné tvoření klíčových slov. A to ve v2.3.4 (jádro 4.2.7) není možné. 
Příklad: První klíčové slovo stáhnu, druhé v databázi nenajdu, takže se vrátím do formuláře, kde volně vytvořím, další klíč. slovo už ale nemůžu stáhnout, protože se v celém formuláři je zašedlý (není aktivní) ten symbol katalogu. 
Po uložení formuláře nebo obnovení změn je sice ikona pro stažení opět zaktivní, ale každé uložení formuláře a opětovné scrollování je pro zpracovatele zdržením. A bohužel v praxi jsme zjistili, že ukládání po přidání každého klíčového slova zpracovatelům nevyhovuje a pro běžnou produkci není použitelné.

Plus otázka na závěr - Bylo by prosím možné po přidání stažené autority do elementu Subject, aby se formulář přesunul do elementu Subject a ne na začátek formuláře?

Souvisí s https://github.com/proarc/proarc-client/issues/690 , které jsem založila na klientovi.

**Comment by SykoraLukas** [comment]
Toto jsou ale dvě různé věci.

1) Editace záznamu ve formuláři

2) Spuštění funkce pro přidání autority nad záznamem (znovu se načte záznam na api, přidá se do něj na poslední místo nová autorita a uloží se. Znovu se načte a ten znovunačtený mods se pošle na klienta, který záznam zobrazí).

Dovedu si představit ale funkci která bude kombinovat funkce "uložit" a "přidat autoritu". Nejprve by záznam uložila, poté by proběhla klasická funkce "přidání autority" a nakonec by se mohlo odskorolovat na poslední přidaný subjekt.

**Comment by SykoraLukas** [comment]
Z mého pohledu toto není chyba, je další návrh na to, jak tuto funkci dále upravit. Mohli bychom na toto založit další issue? To co ta funkce měla dělat tak dělá.

**Comment by tumovam** [comment]
Souhlasím, to, co bylo v zadání funguje. Stažení klíčových slov z báze autorit. Stažené autority mají valueURI i "authority="czenas" . Funguje i Přidat s jazykem. Na klientovi tedy upravím issue 690. Díky
